pennylane/_qubit_device.py

```diff
@@ -269,13 +269,37 @@ def statistics(self, observables):
             elif obs.return_type is Probability:
                 results.append(self.probability(wires=obs.wires))
 
+            elif obs.return_type is State:
+                if len(observables) > 1:
+                    raise ValueError(
+                        "The state cannot be returned in combination with other return types"
+                    )
+                results.append(self._validate_state())
+
             elif obs.return_type is not None:
                 raise QuantumFunctionError(
                     "Unsupported return type specified for observable {}".format(obs.name)
                 )
 
         return results
 
+    def _validate_state(self):
```

Comment:
How about names like `_validate_state_access`, `_can_access_state`? `_validate_state` implies we are validating the state vector

Reply:
Ok, changed to `_access_state()`

Reply:
@trbromley if this is meant to be overridden by inheriting devices (e.g., it is an official part of the plugin API), then it probably shouldn't be private.

I think we have been overusing private methods lately. In Python, there is not really any difference between public and private methods, _except_ that private methods don't currently show up in our Sphinx docs (I believe). Which will make it harder for plugin developers.

Reply:
Done: 2dae0c2a485e91147fa0164f0cd84ae176b88c49